server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -434,6 +437,11 @@ static Settings aggregateIndexSettings(ClusterState currentState, CreateIndexClu
          * that will be used to create this index.
          */
         MetaDataCreateIndexService.checkShardLimit(indexSettings, currentState);
+        if (indexSettings.getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true) == false) {
```

Comment:
Do we want to completely disallow setting this setting in 8.x? If so, should we always emit a warning, even if it's explicitly set to true?

Reply:
Users might be confused about the deprecation when creating indices for CCR in a mixed cluster between 7.x and 6.8 because they need to explicitly set the soft_deletes setting.

Reply:
If mixed clusters with 6.8 is a worry, you can just check here that all nodes are 7.0+.
With the approach here, we will still have to support the setting for the full duration of 8.x (i.e. allow people to explicitly set soft-deletes true).

Reply:
++. I will make this change in a follow-up.